File: SignatureRequests/SignatureRequests.Webs/Scripts/react/src/Entities/RequestEntity.ts
import UserEntity from "./UserEntity";
import BoxResponseList from "./BoxResponseList";
import GroupEntity from "./GroupEntity";

class RequestEntity {
  id: number;
  signer: UserEntity;
  group: GroupEntity | null;
  requestor: UserEntity;
  status: string;
  sentDate: Date;
  boxes: BoxResponseList;

  constructor(data: any) {
    this.id = data.Id;
    this.signer = new UserEntity(data.Signer);
    this.group = data.Group ? new GroupEntity(data.Group) : null;
    this.requestor = new UserEntity(data.Requestor);
    this.status = data.Status;
    this.sentDate = data.SentDate;
    this.boxes = new BoxResponseList(data.Boxes);
  }
}

export default RequestEntity;

File: SignatureRequests/SignatureRequests.Core/Entities/RequestEntity.cs
using BTSuggestions.Core.Entities;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SignatureRequests.Core.Entities
{
    public class RequestEntity: BaseEntity
    {
        [Required]
        public virtual UserEntity Signer { get; set; }

        public int SignerId { get; set; }

        [Required]
        public virtual GroupEntity Group { get; set; }

        public int GroupId { get; set; }
        
        [Required]
        public virtual UserEntity Requestor { get; set; }

        public int RequestorId { get; set; }

        [Required]
        public string Status { get; set; }

        [Required]
        public DateTime SentDate { get; set; }
        public virtual ICollection<BoxEntity> BoxEntities { get; set; }

    }
}
